shard-operator-parent/shard-operator-core/src/test/java/com/redhat/service/smartevents/shard/operator/core/utils/DeploymentSpecUtilsTest.java

```diff
@@ -0,0 +1,28 @@
+package com.redhat.service.smartevents.shard.operator.core.utils;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.redhat.service.smartevents.shard.operator.core.TemplateProvider;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+
+public class DeploymentSpecUtilsTest {
+    @Test
+    public void TestIsDeploymentEqual() {
+        Deployment expectedDeployment = TemplateProvider.loadTestDeployment();
+        Deployment existingDeployment = TemplateProvider.loadTestDeployment();
+        boolean result = DeploymentSpecUtils.isDeploymentEqual(expectedDeployment, existingDeployment);
+        Assertions.assertThat(result).isTrue();
+    }
+
+    @Test
+    public void TestIsDeploymentNotEqual() {
+        Deployment expectedDeployment = TemplateProvider.loadTestDeployment();
+        Deployment existingDeployment = TemplateProvider.loadTestDeployment();
+        existingDeployment.getSpec().getSelector().getMatchLabels().remove("labelkey2");
+        boolean result = DeploymentSpecUtils.isDeploymentEqual(expectedDeployment, existingDeployment);
```

Comment:
This method has [different conditions](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/shard-operator-parent/shard-operator-core/src/main/java/com/redhat/service/smartevents/shard/operator/core/utils/DeploymentSpecUtils.java#L9-L12), can you please add more scenarios (i.e. to test `expected.getSpec().getTemplate().getSpec().getContainers().get(0).getEnv()`).

This equals method is not considering some other fields, it only considers selector, image and env variables as far as I see. Can you please add a test/check where two deployments are different but only for other features to check that `isDeploymentEqual` remains true?